### PR TITLE
feat: implement submission update in form

### LINKED ISSUE #616: Send graphql mutation for saving updatedFacilitySubmissions
# Problem
Now that the form autofills and can show submissions that have been fetched, we need to create a function that allows the form to be submitted and saved with any updated changes.

# Success
Send successful mutations from updated forms.

### Requirements

- [x] Send successful mutations from updated facility forms.

# Implementation notes

### PR DESCRIPTION
**Pull request by NabbeunNabi**
Resolves #616 #637 

## 🔧 What changed
This now implements the graphQL retry from #659 into the update Submission. I have written another ticket to show the UI that it is retrying in the top nav. For now this just adds the functionality where you can mutate the submission. I wanted it to be pulled out as the functionality is more important as a start to get the Mod Panel up and running instead of a countdown message that is fun, but not necessary for `modPanel` MVP

#693 is the follow up ticket that is asking for updating the UI to show the retry countdown similar to the figma: 

## 🧪 Testing instructions
Tested manually by running the backend with (yes yarn seems to work)
```
yarn dev:startlocaldb
yarn dev
```

Then running the frontend with 
```
yarn dev:localserver
```

### CODE REVIEW

File: components/ModEditSubmissionTopbar.vue
Comment (NabbeunNabi) [inline components/ModEditSubmissionTopbar.vue:25]: This retry message appears 3 times if the request fails. To show it is retrying the mutation in 60s with a countdown timer

File: components/ModEditSubmissionTopbar.vue
Comment (NabbeunNabi) [inline components/ModEditSubmissionTopbar.vue:39]: This shows saving text. Very briefly but follows the decision from Figma. If the request takes a while it will show. Depends on speed of server response.

File: components/ModEditSubmissionTopbar.vue
Comment (NabbeunNabi) [inline components/ModEditSubmissionTopbar.vue:34]: sets a state in the `ModSubmissionsStore` that calls the submit function on the `ModEditFacilitySubmissionsStore`

File: components/ModEditSubmissionTopbar.vue
Comment (NabbeunNabi) [inline components/ModEditSubmissionTopbar.vue:75]: uses a created timer that is in a separate file in case we need timers in the future that can use a callback to set other variables in case we need to set states. This one will use a callback that will change a local state of number of seconds and if the retry message should be shown. It is translatable

File: components/ModEditSubmissionTopbar.vue
Comment (NabbeunNabi) [inline components/ModEditSubmissionTopbar.vue:128]: clears the interval on unmounted

File: components/ModSubmissionsEditFormFacility.vue
Comment (NabbeunNabi) [inline components/ModSubmissionsEditFormFacility.vue:394]: this loop handles the retries and state management based on where the file is saved.

File: components/ModSubmissionsEditFormFacility.vue
Comment (NabbeunNabi) [inline components/ModSubmissionsEditFormFacility.vue:421]: this sets up a synthetic event to mock the submit button event when the state changes on click from the Top Nav to save and exit

File: stores/moderationSubmissionsStore.ts
Comment (NabbeunNabi) [inline stores/moderationSubmissionsStore.ts:20]: adds states for manipulation between the ModEdit TopNav and the Form to submit the form

File: cypress/e2e/moderationDashboard.cy.ts
Comment (NabbeunNabi) [inline cypress/e2e/moderationDashboard.cy.ts:390]: Test to make sure the error message for retry is shown on server error sent back

File: utils/countdownTimer.ts
Comment (NabbeunNabi) [inline utils/countdownTimer.ts:1]: Re-useable timer function

File: components/ModEditSubmissionTopbar.vue
Comment (theyokohamalife) [inline components/ModEditSubmissionTopbar.vue:81]: Nit: You could simplify the name to `updatedFormBeingSent`

File: components/ModEditSubmissionTopbar.vue
Comment (theyokohamalife) [inline components/ModEditSubmissionTopbar.vue:83]: Nit: ♻️  `retryCountdown`

File: utils/countdownTimer.ts
Comment (theyokohamalife) [inline utils/countdownTimer.ts:12]: What do you think about initializing `counterInterval` as `0`? That could simplify the type and the conditional could check if it's above `0`.

**Review by theyokohamalife — APPROVED** [review]
Looks good! I left a couple non-blocking comments. You might want to wait for feedback from @ermish as well before merging.

File: i18n/locales/en.json
Comment (ermish) [inline i18n/locales/en.json:129]: I'm a little nervous about the variable being in the translation string here vs just splitting the string into two parts.  Personally, I would change it, but I'll allow it if you prefer this way

Comment (ermish) [reply i18n/locales/en.json:129]: reason: separate strings prevents any mistakes where a future dev might not be aware there's a variable, or forget to put in in one of the translations which would break the code. Explicit sentence parts makes it apparent what is being built.

File: stores/moderationSubmissionsStore.ts
Comment (ermish) [reply stores/moderationSubmissionsStore.ts:20]: when using a state store, you should keep your data abstracted from the usage of it. 
ex. instead of calling data `firstTabData` `secondTabData`, it should be `facilitiesData` `profData`. 

So in this case, we should remove TopNav and  make it clear what it's doing.

File: stores/moderationSubmissionsStore.ts
Comment (ermish) [inline stores/moderationSubmissionsStore.ts:29]: these handles aren't really clear what they're handling. Usually `handle` is used for handling an event. In this case, these just look like `setX` functions

File: stores/moderationSubmissionsStore.ts
Comment (ermish) [inline stores/moderationSubmissionsStore.ts:34]: maybe these could be combine into a `formSubmissionStatus` with `sending`, `failed`, etc.

**Review by ermish — CHANGES_REQUESTED** [review]
dropped a few comments

File: i18n/locales/en.json
Comment (NabbeunNabi) [reply i18n/locales/en.json:129]: When it comes to splitting it would it still easily be clear with seconds being separated from the from part. It is just for the s or 秒

File: stores/moderationSubmissionsStore.ts
Comment (NabbeunNabi) [reply stores/moderationSubmissionsStore.ts:34]: @ermish I will make this a draft again since I am going to wait for the retry logic from the rejection usable function to be implemented

File: components/ModSubmissionsEditFormFacility.vue
Comment (ermish) [inline components/ModSubmissionsEditFormFacility.vue:368]: this should definitely be a `type`

**Review by ermish — CHANGES_REQUESTED** [review]
looking muuuch cleaner after the retry logic updates

File: components/ModSubmissionsEditFormFacility.vue
Comment (NabbeunNabi) [reply components/ModSubmissionsEditFormFacility.vue]: yea you right. Unfortunately since it is a type with the property `id` I cannot use a more descriptive variable name such as `spoon`

**Review by ermish — APPROVED** [review]
ship it!

**Pull request merged**
